### PR TITLE
Fixes #30136: Do not proxy isos or yum on foreman proxy content

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Review by wbclark — CHANGES_REQUESTED** [review]
@ehelms This change in answers would only affect new installations, correct? Therefore for users that upgraded there should also be a migration to disable the proxying of isos and rpm content.

**Comment by ehelms** [comment]
Yessir, you are right. Helps if you git add new files when creating a PR :)

**Comment by jlsherrill** [comment]
@ehelms looks like some rubocop failures

**Comment by ehelms** [comment]
Addressed rubocop @jlsherrill @wbclark

**Comment by jlsherrill** [comment]
LGTM @wbclark ?

File: config/foreman-proxy-content.migrations/200616155948-disable-pulp-3-proxying.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+PROXY_ISOS = 'proxy_pulp_isos_to_pulpcore'.freeze
+
+def upgrade
+  answers[FOREMAN_PROXY_CONTENT] = {} unless answers[FOREMAN_PROXY_CONTENT].is_a?(Hash)
+  answers[FOREMAN_PROXY_CONTENT][PROXY_YUM] = false
+  answers[FOREMAN_PROXY_CONTENT][PROXY_ISOS] = false
+end
+
+if answers[FOREMAN_PROXY_CONTENT] &&
+   answers[FOREMAN_PROXY_CONTENT][PROXY_YUM].nil? &&
```
Comment (wbclark) [inline config/foreman-proxy-content.migrations/200616155948-disable-pulp-3-proxying.rb:13]: I used this pattern previously when I was confident that answers[FPC] should already be a hash, but that wasn't the case for answers in this scenario previously, opening the possibility of a NoMethodError:

```
irb(main):012:0> answers
=> {"FPC"=>true}
irb(main):013:0> if answers['FPC']
irb(main):014:1>   puts "hello"
irb(main):015:1> end
hello
=> nil
irb(main):016:0> if answers['FPC'] && answers['FPC']['PROXY'].nil?
irb(main):017:1>   puts "hello"
irb(main):018:1> end
Traceback (most recent call last):
        2: from lib/tasks/console.rake:5:in `block in <top (required)>'
        1: from (irb):16
NoMethodError (undefined method `[]' for true:TrueClass)
irb(main):019:0> 
```

Comment (wbclark) [reply config/foreman-proxy-content.migrations/200616155948-disable-pulp-3-proxying.rb:13]: Hmmm, my usage may have been incorrect as well. I'm investigating it a bit further.

File: config/foreman-proxy-content.migrations/200616155948-disable-pulp-3-proxying.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+   answers[FOREMAN_PROXY_CONTENT][PROXY_ISOS].nil?
```
Comment (wbclark) [inline config/foreman-proxy-content.migrations/200616155948-disable-pulp-3-proxying.rb]:
```suggestion
if answers[FOREMAN_PROXY_CONTENT] == true ||
   (answers[FOREMAN_PROXY_CONTENT].is_a?(Hash) &&
   answers[FOREMAN_PROXY_CONTENT][PROXY_YUM].nil? &&
   answers[FOREMAN_PROXY_CONTENT][PROXY_ISOS].nil?)
```
I think this works in all cases.

**Review by wbclark — APPROVED** [review]
LGTM. Thanks @ehelms !

**Comment by tbrisker** [comment]
2.1 - 1adc0d7

**Pull request merged**